frontend/src/component/HeatMap.tsx

```diff
@@ -0,0 +1,85 @@
+import {ComposableMap, Geographies, Geography,} from "react-simple-maps";
+import {Country} from "../model/Country";
+import {useEffect, useState} from "react";
+import {scaleLinear} from "d3-scale";
+import useUsers from "../hook/useUsers";
+import {Traveller} from "../model/User";
+
+type HeatMapProps = {
+    countries: Country[]
+}
+const geoUrl = "https://raw.githubusercontent.com/deldersveld/topojson/master/world-countries.json"
+
+export default function HeatMap(props: HeatMapProps) {
+
+    const [maxDomain, setMaxDomain] = useState(1)
+
+    const colorScale = scaleLinear<string>()
+        .domain([0, maxDomain])
+        .range(["#f8bbd0", "#ad1457"]);
+
+    const countryList = props.countries
+    let userList:Traveller[] = useUsers().users
+
+    const [countryCounts, setCountryCounts] = useState<{
+        country: string,
+        count: number
+    }[]>([])
+
+    useEffect(() => {
+        const threeLetterCodeList = countryList.map(Country => Country.threeLetterCode)
+
+        function setCount(country: string) {
+            return {
+                country,
+                count: 0
+            };
+        }
+
+        let output = threeLetterCodeList.map(setCount);
```

Comment:
what is output suppose to mean? maybe u can be a bit more creative with your variable names.